test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -68,48 +82,49 @@ var customLogsConfigTmplFile string
 //go:embed testfixtures/config.yaml.tmpl
 var configTmplFile string
 
-func multiFakeintakeStackDef(agentOptions ...agentparams.Option) *e2e.StackDefinition[multiFakeIntakeEnv] {
-	return e2e.EnvFactoryStackDef(func(ctx *pulumi.Context) (*multiFakeIntakeEnv, error) {
-		awsEnv, err := awsResources.NewEnvironment(ctx)
+func multiFakeIntakeAWS(agentOptions ...agentparams.Option) e2e.Provisioner {
+	runFunc := func(ctx *pulumi.Context, env *multiFakeIntakeEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		vm, err := ec2vm.NewEC2VMWithEnv(awsEnv)
+		host, err := ec2.NewVM(awsEnv, "nssfailover")
```

Comment:
The VM doesn't use a default name like it used to ?

Reply:
Indeed, `name` is a mandatory parameter for any resource, there's no reason not to expose it as such.
A name is provided in pre-built scenarios.